File: src/main/java/kr/co/greentech/measure/domain/Measure.java
package kr.co.greentech.measure.domain;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "MEASURE")
public class Measure {
    @Id
    @GeneratedValue
    @Column(name = "MEASURE_ID")
    private Long measureId;

    private String name;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createTime;

    @Enumerated(EnumType.STRING)
    private MeasureMode mode;

    @Enumerated(EnumType.STRING)
    private MeasureStatus status;


    public static Measure createMeasure(String name, String mode) {
        Measure measure = new Measure();
        measure.setName(name);
        measure.setStatus(MeasureStatus.ING);

        if (mode.equals("step")) {
            measure.setMode(MeasureMode.STEP);
        } else {
            measure.setMode(MeasureMode.INTERVAL);
        }

        measure.setCreateTime(new Date(System.currentTimeMillis()));

        return measure;
    }



    public Long getMeasureId() {
        return measureId;
    }

    public void setMeasureId(Long measureId) {
        this.measureId = measureId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public MeasureMode getMode() {
        return mode;
    }

    public void setMode(MeasureMode mode) {
        this.mode = mode;
    }

    public MeasureStatus getStatus() {
        return status;
    }

    public void setStatus(MeasureStatus status) {
        this.status = status;
    }

}

File: src/main/java/kr/co/greentech/measure/domain/Measure.kt
package kr.co.greentech.measure.domain

import javax.persistence.*

@Entity
@Table(name = "MEASURE")
class Measure {
    @Id
    @GeneratedValue
    @Column(name = "MEASURE_ID")
    var measureId: Long? = null
    var name: String? = null
    var time: Long? = null

    @Enumerated(EnumType.STRING)
    var status: MeasureStatus? = null

    @ManyToOne
    @JoinColumn(name = "COMPANY_ID")
    var company: Company? = null

    companion object {
        @JvmStatic
        fun create(name: String): Measure {
            val measure = Measure()
            measure.name = name
            measure.status = MeasureStatus.ING
            measure.time = System.currentTimeMillis()

            return measure
        }
    }
}